packages/workflow/src/components/workflow-menu.js

```diff
@@ -0,0 +1,302 @@
+		const allAbilities = select( abilitiesStore ).getAbilities();
+		return allAbilities || [];
+	}, [] );
+
+	const filteredAbilities = useMemo( () => {
+		if ( ! search ) {
+			return abilities;
+		}
+		const searchLower = search.toLowerCase();
+		return abilities.filter(
+			( ability ) =>
+				ability.label?.toLowerCase().includes( searchLower ) ||
+				ability.name?.toLowerCase().includes( searchLower )
+		);
+	}, [ abilities, search ] );
+
+	// Focus container when output is shown so it can receive keyboard events
+	useEffect( () => {
+		if ( abilityOutput && containerRef.current ) {
+			containerRef.current.focus();
+		}
+	}, [ abilityOutput ] );
+
+	useEffect( () => {
+		registerShortcut( {
+			name: 'core/workflows',
+			category: 'global',
+			description: __( 'Open the workflow palette.' ),
+			keyCombination: {
+				modifier: 'primary',
+				character: 'j',
+			},
+		} );
+	}, [ registerShortcut ] );
+
+	useShortcut(
+		'core/workflows',
+		/** @type {import('react').KeyboardEventHandler} */
+		( event ) => {
+			// Bails to avoid obscuring the effect of the preceding handler(s).
+			if ( event.defaultPrevented ) {
+				return;
+			}
+
+			event.preventDefault();
+			setIsOpen( ! isOpen );
+		},
+		{
+			bindGlobal: true,
+		}
+	);
+
+	const closeAndReset = () => {
+		setSearch( '' );
+		setIsOpen( false );
+		setAbilityOutput( null );
+		setIsExecuting( false );
+	};
+
+	const goBack = () => {
+		setAbilityOutput( null );
+		setIsExecuting( false );
+		setSearch( '' );
+	};
+
+	const handleExecuteAbility = async ( abilityName ) => {
+		setIsExecuting( true );
+		const ability = abilities.find( ( a ) => a.name === abilityName );
+		try {
+			const result = await executeAbility( abilityName );
+			setAbilityOutput( {
+				abilityName,
+				abilityLabel: ability?.label || abilityName,
+				abilityDescription: ability?.description || '',
+				success: true,
+				data: result,
+			} );
+		} catch ( error ) {
+			setAbilityOutput( {
+				abilityName,
+				abilityLabel: ability?.label || abilityName,
+				abilityDescription: ability?.description || '',
+				success: false,
+				error: error.message || String( error ),
+			} );
+		} finally {
+			setIsExecuting( false );
+		}
+	};
+
+	const onKeyDown = ( event ) => {
+		if (
+			// Ignore keydowns from IMEs
+			event.nativeEvent.isComposing ||
+			// Workaround for Mac Safari where the final Enter/Backspace of an IME composition
+			// is `isComposing=false`, even though it's technically still part of the composition.
+			// These can only be detected by keyCode.
+			event.keyCode === 229
+		) {
+			event.preventDefault();
+		}
+	};
+
+	const onContainerKeyDown = ( event ) => {
+		// Handle going back when viewing output
+		if (
+			abilityOutput &&
+			( event.key === 'Escape' ||
+				event.key === 'Backspace' ||
+				event.key === 'Delete' )
+		) {
+			event.preventDefault();
+			event.stopPropagation();
+			goBack();
+		}
+	};
+
+	if ( ! isOpen ) {
+		return false;
```

Comment:
Nit pick: When a component renders nothing normally we return null instead of false.